[model] support mistralai/Devstral-Small-2505

hieuchi911 opened the pull request:
# PR type
- [ ]  Bug Fix
- [ ]  New Feature
- [ ]  Document Updates
- [x] More Models or Datasets Support

# PR information

- Register devstral model, providing custom function `devstral get_model_tokenizer_devstral_2505` as `get_function`, as `mistralai/Devstral-Small-2505` does not have huggingface tokenizers. In this PR, `mistralai/Mistral-Small-3.1-24B-Instruct-2503` tokenizer is used, as the two models use the same tokenizer.
- Register devstral prompt template
- Add mistral agent template for tool-calling

## Experiment results

Paste your experiment result here(if needed).

Jintao-Huang commented on examples/train/rft/rft.py:
```diff
@@ -22,7 +22,7 @@ def do_sample(model: str, model_type: str, dataset: List[str], iter: int):
         sample_cmd = (f'{conda_prefix} USE_OPENCOMPASS_EVALUATOR=True CUDA_VISIBLE_DEVICES={device} swift sample '
                       f'--model {model} --model_type {model_type} '
-                      f'--dataset {" ".join(dataset)} '
+                      f'--dataset {' '.join(dataset)} '
```
Hello, there is a syntax error here. Please revert this section.

Jintao-Huang commented:
Please pass the lint test, referring to here:
https://github.com/modelscope/ms-swift/blob/main/CONTRIBUTING.md#code-standards-and-development-approach

hieuchi911 commented:
@Jintao-Huang I've added a test case for devstral. Regarding the lint test, commit `c44d7b9` contains changes from running `pre-commit run --all-files` , all of which tried to swap double quotes with single quote like what you [pointed out](https://github.com/modelscope/ms-swift/pull/5102#discussion_r2230075557), so that's why I reverted in commit `8c6608a`. I'm not sure how I should process

Jintao-Huang commented:
Can you run it with Python 3.10? There's a syntax error.

Nortinwell commented:
Does this support voxtral-mini-3B fine-tuning?
